src/main/java/com/firebolt/jdbc/resultset/column/ColumnType.java

```diff
@@ -108,6 +111,8 @@ private static List<ColumnType> getCollectionSubType(FireboltDataType fireboltDa
 		if (fireboltDataType.equals(TUPLE)) {
 			types = typeWithoutNullKeyword.split(",(?![^()]*\\))"); // Regex to split on comma and ignoring comma that are between
 			// parenthesis
+		} else if (fireboltDataType.equals(STRUCT)) {
+			types = typeWithoutNullKeyword.split(",");
```

Comment:
I believe this will fail for nested structs, for example
```
struct(a int, s struct(a int, b text))
typeWithoutNullKeyword.split(",") = "a int, s struct(a int, b text)".split(",") = ["a int", "s struct(a int", "b text)"]
```

Reply:
Well spotted. Looks like it's already taken care of in regex above so I've stolen it.